experimental/sdk/webpubsub-socketio-extension/src/SIO/components/web-pubsub-adapter.ts

```diff
@@ -188,13 +196,70 @@ groupNames = ${toString(rooms)}, connectionId(eioSid) = ${this._getEioSid(id)}`)
    * @param clientCountCallback - the number of clients that received the packet
    * @param ack - the callback that will be called for each client response
    */
-  public broadcastWithAck(
+  public async broadcastWithAck(
     packet: SioPacket,
     opts: BroadcastOptions,
     clientCountCallback: (clientCount: number) => void,
     ack: (...args: unknown[]) => void
-  ): void {
-    throw NotImplementedError;
+  ): Promise<void> {
+    debug(`broadcastWithAck, start, packet = ${JSON.stringify(packet)},\
+  opts = ${toOptionsString(opts)}, namespace = "${this.nsp.name}"`);
+
+    let accumulatedData = "";
+    let count = 0;
+
+    const streamHandleResponse = (chunk: string) => {
+      const handleJsonLines = (lines: string[], onPacket: (SioPacket) => void) => {
+        /**
+         * Line 1: {xxx}
+         * Line 2: {xxx}
+         * ..
+         * Line N: {xx   // maybe not complete
+         */
+        for (let i = 0; i < lines.length - 1; i++) {
+          if (lines[i]) {
+            const emitWithAckResponse = JSON.parse(lines[i]);
+            // The payload is UTF-8 encoded EIO payload, we need to decode it and only ack the data
+            const eioPackets = EioParser.decodePayload(emitWithAckResponse.Payload);
+            this._sioDecoder.on("decoded", (packet: SioPacket) => onPacket(packet));
+            eioPackets.forEach((element) => {
+              this._sioDecoder.add(element.data);
+            });
+            this._sioDecoder.off("decoded");
+          }
+        }
+      };
+
+      accumulatedData += chunk.toString();
+      const lines = accumulatedData.split("\n");
+      handleJsonLines(lines, (packet: SioPacket) => {
+        ack(...packet.data);
+        count++;
+      });
+      accumulatedData = lines[lines.length - 1];
+    };
+
+    const bodyHandler: (data: Uint8Array | undefined, end: boolean) => void = (value, end) => {
+      if (end) {
```

Comment:
Seems we can validate data `if (data) {}` first as we can't make sure data is undefined when `end is true`